### PR TITLE
[RPA 804] Create CRON job to delete saved guest accounts.

### PR DESCRIPTION
**Pull request by MRSIMM0**
## Description

Created CRON job to delete all gust accounts every day at midnight.

Added trigger to delete processes of those guests. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

---

I agree to the terms of the RunBotics Contributor License Agreement.

### CODE REVIEW

File: runbotics-orchestrator/src/main/java/com/runbotics/web/rest/GuestResource.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+     * {@code DELETE /api/guest/delete/all} : delete the "login" User.
+     *
+     * @return the {@link ResponseEntity} with status {@code 204 (NO_CONTENT)}.
+     */
+    @DeleteMapping("/delete/all")
+    @PreAuthorize("hasAuthority(\"" + AuthoritiesConstants.ADMIN + "\")")
+    public ResponseEntity<Void> deleteUser() {
+        log.debug("REST request to delete All Guest Accounts");
+        guestService.deleteAllGuestAccounts();
+        return ResponseEntity.noContent().headers(HeaderUtil.createAlert(applicationName, "guestManagement.deleted","All")).build();
```
Comment (star-in-the-jar) [inline runbotics-orchestrator/src/main/java/com/runbotics/web/rest/GuestResource.java]: format - add NL

File: runbotics-orchestrator/src/main/java/com/runbotics/web/rest/GuestResource.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    private GuestService guestService;
+
+    @Value("${jhipster.clientApp.name}")
+    private String applicationName;
+    public GuestResource(GuestService guestService) {
+        this.guestService = guestService;
+    }
+
+    /**
+     * {@code DELETE /api/guest/delete/all} : delete the "login" User.
```
Comment (star-in-the-jar) [inline runbotics-orchestrator/src/main/java/com/runbotics/web/rest/GuestResource.java]: > delete the "login" User.

i belive it "deletes all guests"

**Review by star-in-the-jar — CHANGES_REQUESTED** [review]
nice

**Review by star-in-the-jar — DISMISSED** [review]

**Review by star-in-the-jar — APPROVED** [review]

File: runbotics-orchestrator/src/main/java/com/runbotics/web/rest/GuestResource.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import tech.jhipster.web.util.HeaderUtil;
+
+import javax.validation.constraints.Pattern;
+
+@RestController
+@RequestMapping("/api/guest")
```
Comment (preludium) [inline runbotics-orchestrator/src/main/java/com/runbotics/web/rest/GuestResource.java:20]: guests*. API resources should be in the plural form

File: runbotics-orchestrator/src/main/java/com/runbotics/web/rest/GuestResource.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    public GuestResource(GuestService guestService) {
+        this.guestService = guestService;
+    }
+
+    /**
+     * {@code DELETE /api/guest/delete/all} : delete all guest accounts.
+     *
+     * @return the {@link ResponseEntity} with status {@code 204 (NO_CONTENT)}.
+     */
+    @DeleteMapping("/delete/all")
```
Comment (preludium) [inline runbotics-orchestrator/src/main/java/com/runbotics/web/rest/GuestResource.java:38]: the information about the delete is in the endpoint mapping so leave just `/all`

**Pull request merged**